timemachine/fe/topology.py

```diff
@@ -144,46 +144,73 @@ def parameterize_periodic_torsion(self, proper_params, improper_params):
         )
         return self._parameterize_bonded_term(guest_params, guest_potential, self.host_periodic_torsion)
 
-    def parameterize_nonbonded(self, ff_q_params, ff_q_params_intra, ff_lj_params, lamb: float):
+    def parameterize_nonbonded(self, ff_q_params, ff_q_params_intra, ff_q_params_solv, ff_lj_params, lamb: float):
         num_guest_atoms = self.guest_topology.get_num_atoms()
-        guest_params, guest_pot = self.guest_topology.parameterize_nonbonded(
-            ff_q_params, ff_q_params_intra, ff_lj_params, lamb, intramol_params=False
+        # ligand other interactions
+        guest_ixn_other_params, _ = self.guest_topology.parameterize_nonbonded(
+            ff_q_params, None, None, ff_lj_params, lamb, intramol_params=False
         )
 
-        assert guest_params.shape == (num_guest_atoms, 4)
-        assert self.host_nonbonded is not None
-        assert guest_pot.beta == self.host_nonbonded.potential.beta
-        assert guest_pot.cutoff == self.host_nonbonded.potential.cutoff
+        # ligand water interactions
+        guest_ixn_water_params, _ = self.guest_topology.parameterize_nonbonded(
```

Comment:
By passing this in the first place, it assumes that the handlers of the solv/non-solv are identical. While currently true, may not be true long term. At least a comment would be good.